### ISSUE TITLE
No support for multi-line MUSTACHE templates in initializerJson.json

### ISSUE THREAD

**Issue by Arc-E-Tect** [mock-server/mockserver#1840]
**Describe the issue**
I am using the initializerJson.json to initialize Mock-Server. For my team it is more convenient to manage a standard set of mock-responses this way.

When specifying literal request/responses we can use definitions that are easy to format and read, but when using templates this is cumbersome and the documentation is not very helpful.

**What you are trying to do**
Trying to define a MUSTACHE based template in initializerJson.json file as

```json
[{
    "httpRequest": {
        "method": "GET",
        "path": "/persons/{id}",
        "pathParameters": {
            "id": [
                "[0-9\\-]+"
            ]
        }
    },
    "httpResponseTemplate": {
        "templateType": "MUSTACHE",
        "template": {
            "statusCode": 200,
            "headers": {
                "Content-Type": [
                    "application/json"
                ]
            },
            "body": {
                "name": "John",
                "id": "{{path.id}}"
            }
        }
    }
}]
```

but I have to specify it as:
```json
[{
    "httpRequest": {
      "method": "GET",        
      "path": "/persons/{id}",
      "pathParameters": {
            "id": ["[0-9\\-]+"]
        }
    },
    "httpResponseTemplate": {
        "templateType": "MUSTACHE",
        "template": "{\"statusCode\":200,\"headers\":{\"Content-Type\":[\"application/json\"]},\"body\":{\"name\":\"John\",\"id\":\"{{path.id}}\"}}"
    }
}]
```
Note that the template field is one long hard to maintain string.

Something tells me that there must be a way for a more convenient way to specify this.

**MockServer version**
5.15

**To Reproduce**
Use the snippets above as initializerJson,json file and the first snippet will result in an error

1. How you are running MockServer (i.e maven plugin, docker, etc): Docker
2. Code you used to create expectations: initializerJson.json
3. What error you saw: exception while loading JSON initialization file with file watcher, ignoring file

**Expected behaviour**
What you expected to happen: support for friendly format of json when using templates

**MockServer Log**
Log output, as INFO level (or lower)

**Comment by markieww** [comment]
I am using the "Language Inject"- and "Edit Code Fragment"-feature of Jetbrains Intellij IDEA to circumvent this JSON-restriction.

Here is an example (GIF-based screencast)
![Image](https://github.com/user-attachments/assets/c04877d8-b8a0-47da-9201-573ec9f974fd)

See the documentation at https://www.jetbrains.com/help/idea/using-language-injections.html

**Comment by freudl** [comment]
Indeed, it would really help if [`template`](https://app.swaggerhub.com/apis/jamesdbloom/mock-server-openapi/5.15.x#/HttpTemplate) accepted more than bare strings.

**Comment by jamesdbloom** [comment]
Closing — this is by design rather than a bug. The `template` field must be a string because it contains Mustache syntax (e.g. `{{path.id}}`) that needs to be processed as a template engine input, not as a JSON structure.

**Workaround for readability:** You can use JSON string concatenation or a multi-line string approach depending on your tooling. In the initializer JSON, you can also use `\n` for newlines within the string to keep the template somewhat readable:

```json
"template": "{\n  \"statusCode\": 200,\n  \"body\": {\n    \"id\": \"{{path.id}}\"\n  }\n}"
```

Alternatively, use the MockServer client API (Java, JavaScript, etc.) to create expectations programmatically, which avoids the JSON escaping entirely. The response templates documentation at https://www.mock-server.com/mock_server/response_templates.html covers this approach.
